### PR TITLE
Real parser

### PR DESCRIPTION
**Pull request by jaredpar**

### CODE REVIEW

File: src/Compilers/Core/Portable/RealParser.cs [hunk 1]
```diff
@@ -0,0 +1,806 @@
...
+namespace Microsoft.CodeAnalysis
+{
+#if OS_UNIX
+
+    // TODO: https://github.com/dotnet/roslyn/issues/5870
+    internal static class RealParser
+    {
+        public static bool TryParseDouble(string s, out double d)
+        {
+            return double.TryParse(s, out d);
```
Comment (gafter) [inline src/Compilers/Core/Portable/RealParser.cs]: Do the tests pass on Unix with this implementation?

Comment (jaredpar) [inline src/Compilers/Core/Portable/RealParser.cs]: @gafter bah, I didn't think about that.  I just ran the build on my machine until I confirmed the bootstrap process passed.  If any tests fail on Unix I'll disable them for Unix runs only (referencing the tracking bug).

Comment (gafter) [inline src/Compilers/Core/Portable/RealParser.cs]: Yeah, the tests that check for correctness of the floating-point scanning code should fail.

**Comment by jaredpar** [comment]
CC @gafter

**Comment by jaredpar** [comment]
test win 32 please

File: src/Compilers/Core/Portable/RealParser.cs [hunk 2]
```diff
@@ -0,0 +1,786 @@
...
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// A set of utilities for converting from a decimal floating-point literal string to its IEEE float
+    /// or double representation, which coniders all digits signficant and correctly rounds according to
+    /// the IEEE round-to-nearest-ties-to-even mode. This code does not support a leading sign character,
+    /// as that is not part of the C# or VB floating-point literal lexical syntax.
+    /// 
+    /// If you change this code, please run the set of long-running random tests in the solution
+    /// RandomRealParserTests.sln. That solution is not included in Roslyn.sln as it is Windows-specific.
```
Comment (dpoeschl) [inline src/Compilers/Core/Portable/RealParser.cs]: > That solution is not included in Roslyn.sln as it is Windows-specific.

Hmm?

Comment (gafter) [inline src/Compilers/Core/Portable/RealParser.cs]: It is a separate .sln (Solution) that is written using Windows-specific facilities.

**Pull request merged**